### PR TITLE
Shut down Servo when the ML app quits

### PR DESCRIPTION
**Pull request by asajeffrey**
Shut down servo when shutting down the magic leap app.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because we don't test ML

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22094)

### CODE REVIEW

File: ports/libmlservo/src/lib.rs [hunk 1]
```diff
@@ -221,11 +224,25 @@ pub unsafe extern "C" fn navigate_servo(servo: *mut ServoInstance, text: *const
...
 #[no_mangle]
 pub unsafe extern "C" fn discard_servo(servo: *mut ServoInstance) {
-    // Servo drop goes here!
-    if !servo.is_null() {
-        Box::from_raw(servo);
+    if let Some(servo) = servo.as_mut() {
+        let mut servo = Box::from_raw(servo);
+        let finish = Instant::now() + SHUTDOWN_DURATION;
+        'outer: while Instant::now() < finish {
+            servo.servo.handle_events(vec![WindowEvent::Quit]);
```
Comment (paulrouget) [inline ports/libmlservo/src/lib.rs]: Send only one quit event and then call `handle_events([])`.
And print a warning/error if it reaches the SHUTDOWN_DURATION limit.

Comment (asajeffrey) [reply ports/libmlservo/src/lib.rs]: Done.

**Comment by asajeffrey** [comment]
Ah, the needs-squash label doesn't block merging, oops.

**Pull request merged**